app/models/spree/gateway/stripe_gateway.rb

```diff
@@ -74,6 +81,23 @@ def create_profile(payment)
 
     private
 
+    def non_fractional_currency?(currency)
+      self.currencies_without_fractions.include?(currency.to_s)
+    end
+
+    def localize_money(money, gateway_options)
+      currency = gateway_options[:currency]
+      # We have this hack in place because in `activemerchant` the library
+      # automatically divides the number by 100, therefore we end up charging
+      # the customer much less than what we intened to do
+      # Example: pass 3000JPY, divide by 100, try to charge 30JPY.
+      if non_fractional_currency?(currency)
+        money * 100
+      else
+        money
```

Comment:
Does this end up working fine for currencies with `subunit_to_unit` values of 1000 and 10000?